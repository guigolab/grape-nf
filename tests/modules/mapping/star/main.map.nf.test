nextflow_process {

    name "Test Process map"
    script "modules/mapping/star/main.nf"
    process "map"

    tag "module"

    setup {
        run("index") {
            script "modules/mapping/star/main.nf"
            process {
                """
                input[0] = file("${baseDir}/data/genome.fa")
                input[1] = file("${baseDir}/data/annotation.gtf")
                """
            }
        }
    }

    test("Should align reads to the reference with compressed annotation") {

        when {
            params {
                mappingTool  = "STAR"
                sjOverHang = 100
                quantificationTool = "RSEM"
                maxMismatches = 4
                maxMultimaps = 10
            }
            process {
                """
                input[0] = file("${baseDir}/data/annotation.gtf.gz")
                input[1] = index.out
                input[2] = Channel.from(
                    [
                        [ "sample1", "test1", file("${baseDir}/data/test1_*.fastq.gz"), "fastq", ["fqRd1", "fqRd2"], '']
                    ]
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.genomeAlignments,
                process.out.genomeAlignmentsIndices,
                process.out.transcriptomeAlignments,
                process.out.junctions
            ).match()
        }

    }

    test("Should align reads to the reference with uncompressed annotation") {

        when {
            params {
                mappingTool  = "STAR"
                sjOverHang = 100
                quantificationTool = "RSEM"
                maxMismatches = 4
                maxMultimaps = 10
            }
            process {
                """
                input[0] = file("${baseDir}/data/annotation.gtf")
                input[1] = index.out
                input[2] = Channel.from(
                    [
                        [ "sample1", "test1", file("${baseDir}/data/test1_*.fastq.gz"), "fastq", ["fqRd1", "fqRd2"], '']
                    ]
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.genomeAlignments,
                process.out.genomeAlignmentsIndices,
                process.out.transcriptomeAlignments,
                process.out.junctions
            ).match()
        }

    }

}
