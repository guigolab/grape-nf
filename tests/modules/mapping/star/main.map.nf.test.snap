{
    "Should align reads to the reference using STAR (uncompressed annotation)": {
        "content": [
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toGenome.bam:md5,4f919efd42c545e1ebdbb8a9780097e8",
                    "bam",
                    "GenomeAlignments",
                    true
                ]
            ],
            [
                
            ],
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toTranscriptome.bam:md5,e6a72cca10b1ac084ff2ae4f572987b8",
                    "bam",
                    "TranscriptomeAlignments",
                    true
                ]
            ],
            [
                [
                    "sample1",
                    "test1",
                    "SJ.out.tab:md5,ec651e0f3bc8338345aa15c8f74038c2",
                    "tsv",
                    "STARjunctions",
                    true
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-23T23:19:28.403716"
    },
    "Should align reads to the reference using STAR (compressed annotation)": {
        "content": [
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toGenome.bam:md5,4f919efd42c545e1ebdbb8a9780097e8",
                    "bam",
                    "GenomeAlignments",
                    true
                ]
            ],
            [
                
            ],
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toTranscriptome.bam:md5,e6a72cca10b1ac084ff2ae4f572987b8",
                    "bam",
                    "TranscriptomeAlignments",
                    true
                ]
            ],
            [
                [
                    "sample1",
                    "test1",
                    "SJ.out.tab:md5,ec651e0f3bc8338345aa15c8f74038c2",
                    "tsv",
                    "STARjunctions",
                    true
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-23T23:11:13.889531"
    }
}